include/matrix_operations/EigenvaluesProblems.h

```diff
@@ -287,6 +287,40 @@ class EigenvaluesProblems<NT, Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic>, E
         return lambdaMinPositive;
     }
 
+    /// Find the minimum positive and maximum negative eigenvalues of the generalized eigenvalue
+    /// problem A + lB, where A, B symmetric and A negative definite.
+    /// \param[in] A Input matrix
+    /// \param[in] B Input matrix
+    /// \return The pair (minimum positive, maximum negative) of eigenvalues
+    NT minPosLinearEigenvalue(MT const & A, MT const & B, VT &eigvec) const
+    {
+        int matrixDim = A.rows();
+        double lambdaMinPositive;
```

Comment:
you can avoid declaration here (is it always `double` ?) and use `auto` in initialization below.